### PR TITLE
Upgrade terraform-provider-azurerm to v4.2.0

### LINKED ISSUE #2349: Upgrade terraform-provider-azurerm to v4.0.1
Release details: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v4.0.1

### LINKED ISSUE #2379: Upgrade terraform-provider-azurerm to v4.1.0
Release details: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v4.1.0

### LINKED ISSUE #2394: Upgrade terraform-provider-azurerm to v4.2.0
Release details: https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v4.2.0

### PR DESCRIPTION
**Pull request by danielrbradley**
This PR was generated via `$ upgrade-provider pulumi/pulumi-azure --upstream-provider-name=terraform-provider-azurerm --kind=provider --target-version=4.2.0`.

---

- Upgrading terraform-provider-azurerm from 4.1.0  to 4.2.0.
	Fixes #2394
	Fixes #2379
	Fixes #2349

### CODE REVIEW

File: patches/0012-Ensure-servicebus-queue-topic-have-namespaces-and-RG.patch [hunk 1]
```diff
@@ -7,19 +7,19 @@ Subject: [PATCH] Ensure servicebus queue & topic have namespaces and RG names
 These are computed attributes for use in the servicebus mixins in Pulumi.
 
 diff --git a/internal/services/servicebus/servicebus_queue_resource.go b/internal/services/servicebus/servicebus_queue_resource.go
-index 579c5712ce..cc468319b5 100644
+index b4df3f33be..fc4838204d 100644
 --- a/internal/services/servicebus/servicebus_queue_resource.go
 +++ b/internal/services/servicebus/servicebus_queue_resource.go
-@@ -47,6 +47,8 @@ func resourceServiceBusQueue() *pluginsdk.Resource {
+@@ -46,6 +46,8 @@ func resourceServiceBusQueue() *pluginsdk.Resource {
```
Comment (t0yv0) [inline patches/0012-Ensure-servicebus-queue-topic-have-namespaces-and-RG.patch:13]: Just line # churn?

File: provider/cmd/pulumi-resource-azure/bridge-metadata.json [hunk 2]
```diff
@@ -13153,25 +13161,6 @@
                 "current": "azure:marketplace/roleAssignment:RoleAssignment",
                 "majorVersion": 6
             },
-            "azurerm_media_services_account_filter": {
```
Comment (t0yv0) [inline provider/cmd/pulumi-resource-azure/bridge-metadata.json:13156]: gone?

**Review by t0yv0 — APPROVED** [review]
LGTM if removals are expected. Thanks!

File: provider/cmd/pulumi-resource-azure/bridge-metadata.json (hunk 2 above)
Comment (danielrbradley) [reply provider/cmd/pulumi-resource-azure/bridge-metadata.json:13156]: This was meant to be removed in the major version, but was accidentally not deleted: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#azurerm_media_services_account_filter

File: patches/0012-Ensure-servicebus-queue-topic-have-namespaces-and-RG.patch (hunk 1 above)
Comment (danielrbradley) [reply patches/0012-Ensure-servicebus-queue-topic-have-namespaces-and-RG.patch:13]: Yup, think so - the there's no diff of the actual line changes

**Pull request merged**